### ISSUE TITLE
time out while initial server greeting

### ISSUE THREAD

**Issue by s0ftcorn** [docker-mailserver/docker-mailserver#3094]
# Subject

Other

## Description
Basically i get this for nearly every mailserver im trying to send mail to:

`Feb 16 19:55:31 mail postfix/smtp[1469]: 421DDF840E6: to=<redacted@gmx.de>, relay=mx00.emig.gmx.net[212.227.15.9]:25, delay=14142, delays=13542/0.04/600/0, dsn=4.4.2, status=deferred (conversation with mx00.emig.gmx.net[212.227.15.9] timed out while receiving the initial server greeting)
´

I got ufw with ufw-docker, allowed the container via `ufw-docker allow mailserver` and connecting via telnet works:
`
$ telnet mx00.emig.gmx.net 25
Trying 212.227.15.9...
Connected to mx00.emig.gmx.net.
Escape character is '^]'.
^]
telnet> quit
Connection closed.
´
My mailserver is not on any blocklist. SPF, DMARC and DKIM are correct. reverse DNS record is correct. Inbound mail is working fine. Two other mailservers with nearly identical setups are working perfectly fine tho.

Logs:
`
$ docker logs mailserver -f
[   INF   ]  Welcome to docker-mailserver 11.3.0
[   INF   ]  Initializing setup
[   INF   ]  Checking configuration
[   INF   ]  Configuring mail server
[   INF   ]  Post-configuration checks
[   INF   ]  Starting miscellaneous tasks
[   INF   ]  Starting daemons
[   INF   ]  mail.stoll-cloud.de is up and running
Feb 16 19:45:28 mail amavis[901]: starting. /usr/sbin/amavisd-new at mail.stoll-cloud.de amavisd-new-2.11.1 (20181009), Unicode aware, LC_CTYPE="C.UTF-8"
Feb 16 19:45:28 mail amavis[901]: perl=5.032001, user=, EUID: 110 (110);  group=, EGID: 112 112 (112 112)
Feb 16 19:45:29 mail amavis[901]: Net::Server: Group Not Defined.  Defaulting to EGID '112 112'
Feb 16 19:45:29 mail amavis[901]: Net::Server: User Not Defined.  Defaulting to EUID '110'
Feb 16 19:45:29 mail amavis[901]: No ext program for   .zoo, tried: zoo
Feb 16 19:45:29 mail amavis[901]: No ext program for   .doc, tried: ripole
Feb 16 19:45:29 mail amavis[901]: No decoder for       .F   
Feb 16 19:45:29 mail amavis[901]: No decoder for       .doc 
Feb 16 19:45:29 mail amavis[901]: No decoder for       .zoo 
Feb 16 19:45:31 mail postfix/master[1448]: daemon started -- version 3.5.17, configuration /etc/postfix
Feb 16 19:45:31 mail postfix/qmgr[1452]: 040C3F807C0: from=<admin@domain.de>, size=1443, nrcpt=1 (queue active)
´
All of this is hosted on ubuntu 22.04 and:
`
$ docker -v
Docker version 23.0.1, build a5ee5b1
´

Anyone of you maybe got a idea? Or what i could try?

**Comment by polarathene** [comment]
You can search our issues for `ufw`, I am pretty sure there was at least one issue discussing some compatibility issue with that software (_or firewalls in general?_). Might be worth looking into.

> Anyone of you maybe got a idea? Or what i could try?

You could try disabling `ufw` briefly?

**Comment by s0ftcorn** [comment]
Thank you for your input. I tried that and the result stays the same:
`Feb 16 19:51:30 mail postfix/smtp[1678]: 6064CF840F4: conversation with mx00.emig.gmx.net[212.227.15.9] timed out while receiving the initial server greeting`

**Comment by s0ftcorn** [comment]
Seems to be a configuration error either on my end or the hosters. I transferred DMS to another host at a different provider and the exact same DMS config works flawless. 
Outside of UFW i have no idea what could be the cause of this, but then when i disable UFW the problem persists.

**Comment by polarathene** [comment]
Oh that is odd. Perhaps it's something else to do with the host OS? Maybe iptables / nftables config differs?

Oh I just noticed this was about sending mail outbound? If that's via Port 25, it's not uncommon for that to be blocked by the provider. Your new host I guess is allowing you to send mail via port 25.

The reason for it being commonly restricted is to deter spam abuse, often requiring you to request access from the provider, or relaying through a smart host service like SendGrid.

If I've understood the issue correctly, did you contact your provider about that at all?
